External/Plugins/ASCompletion/Model/PathModel.cs

```diff
@@ -41,19 +41,14 @@ static public void ClearAll()
         /// </summary>
         static public void Compact()
         {
-            lock (pathes)
```

Comment:
This is the only part of the code locking `pathes`, so it doesn't make sense at all.

Reply:
Actually after second thought, even if the whole `pathes` thing is not fully thread-safe this code adds some safety to the `Compact` function because of the assignment at the end, but that's something easily solvable without fully locking.

I will try to commit a change here ASAP. Anyway, @elsassph, it would be great if you could check all this stuff in the meantime.

Reply:
Yes I used `pathes` as lock object - maybe you should wrap in a `lock (lockObject)` section.

Reply:
That would be wrong, lockObject is specific to each PathModel instance, and these are static fields. I guess every pathes access should be locked, not just this one... but since I never saw any error with it I guess it's not that much of a problem?

Reply:
I decided to put back the locking over `pathes`. Making this part of the logic thread-safe without locking is simple, but I don't feel comfortable doing so without checking the rest of parts using `pathes`.